core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java

```diff
@@ -88,6 +88,14 @@
    Configuration defineConfiguration(String cacheName, String templateCacheName, Configuration configurationOverride);
 
    /**
+    * Removes a configuration from the set of defined configurations. If the named configuration does not exist, nothing
+    * happens
+    *
+    * @param configurationName     the named configuration
+    */
+   void undefineConfiguration(String configurationName);
```

Comment:
We should add a note in the Javadoc here about the IllegalStateException, we could even add it as throws in the method declaration.